tests/models/test_fp8kv_flashinfer.py

```diff
@@ -0,0 +1,96 @@
+
+import pytest
+from transformers import AutoTokenizer
+
+from tests.quantization.utils import is_quant_method_supported
+from vllm import LLM, SamplingParams
+
+os.environ["TOKENIZERS_PARALLELISM"] = "true"
+
+MAX_MODEL_LEN = 1024
+
+MODELS = [
+    "nm-testing/Meta-Llama-3-8B-Instruct-FP8",
+]
+
+EXPECTED_STRS_MAP = {
+    "nm-testing/Meta-Llama-3-8B-Instruct-FP8": {
+        "auto": [
+            'LLaMA is a high-throughput and memory-efficient inference and serving engine for Large Language Models (',
+            'Here are the major milestones in the development of artificial intelligence (AI) from 1950 to ',
+            'Artificial intelligence (AI) and human intelligence (HI) differ significantly in how they process information.',
+            'A neural network is a complex system modeled after the human brain, consisting of interconnected nodes or "ne',
+            'In the sterile, metallic halls of the robotics lab, a peculiar phenomenon occurred. Zeta-5',
+            'The COVID-19 pandemic has had a profound impact on global economic structures and future business models. The',
+            'The Mona Lisa, painted by Leonardo da Vinci in the early 16th century, is one of',
+            'Here are the translations:\n\n**Japanese:** (Haya aki no tori, mushi o',
+        ],
+        "fp8": [
+            'LLM (Large Language Model) is a type of artificial intelligence (AI) model that is trained',
+            'Here are the major milestones in the development of artificial intelligence (AI) from 1950 to ',
+            'Artificial intelligence (AI) and human intelligence (HI) differ significantly in how they process information.',
+            'A neural network is a complex system modeled after the human brain, composed of interconnected nodes or "ne',
+            'Zeta-5, a highly advanced robot designed for menial labor, whirred and beep',
+            'The COVID-19 pandemic has had a profound impact on global economic structures and future business models. Here',
+            'The Mona Lisa, painted by Leonardo da Vinci in the early 16th century, is one of',
+            'Here are the translations:\n\n**Japanese:** (Haya aki no tori, guri o',
+        ]
+    }
+}
+
+
+# This test compares against golden strings for exact match since
+# there is no baseline implementation to compare against
+# and is unstable w.r.t specifics of the fp8 implementation or
+# the hardware being run on.
+# No assert to prevent it from breaking the build
+@pytest.mark.skipif(not is_quant_method_supported("fp8"),
+                    reason="fp8 is not supported on this GPU type.")
+@pytest.mark.parametrize("model_name", MODELS)
+@pytest.mark.parametrize("kv_cache_dtype", ["auto", "fp8"])
+@pytest.mark.parametrize("backend", ["XFORMERS", "FLASHINFER"])
+def test_models(example_prompts, model_name, kv_cache_dtype, backend) -> None:
+    # Note that the golden strings may not work for FLASHINFER Backend.
+    # The intention is to test the path
```

Comment:
In this case I'd suggest compare the logprobs of a first few generated tokens. Otherwise this test will slowdown the CI due to relative large model (8B) and large max tokens.

Reply:
Can you please clarify? I am seeing no output for logprobs.
```
outputs=[CompletionOutput(index=0, text='LLaMA is a high-throughput and memory-efficient inference and serving engine for Large Language Models (', token_ids=array('l', [4178, 64, 4940, 374, 264, 1579, 43847, 631, 323, 5044, 73916, 45478, 323, 13788, 4817, 369, 20902, 11688, 27972, 320]), cumulative_logprob=None, logprobs=None, finish_reason=length, stop_reason=None)], finished=True, metrics=RequestMetrics(arrival_time=1725049625.3656614, last_token_time=1725049625.3656614, first_scheduled_time=1725049625.368224, first_token_time=1725049625.3892546, time_in_queue=0.0025625228881835938, finished_time=1725049625.5246484, scheduler_time=0.0011748597025871277, model_forward_time=None, model_execute_time=None), lora_request=None)
```
Do we have to explicitly enable or disable something in the generate call?

Reply:
Given that the test won't fail, we merge this PR to fix the CI first. I'll have a follow-up PR to improve this test next week.